### ISSUE TITLE
add support for excluding full directories with excludeContracts

### ISSUE THREAD

**Issue by thegostep** [cgewecke/hardhat-gas-reporter#50]

**Comment by cgewecke** [comment]
Thanks for opening. 

It looks like the `excludeContracts` option was completely broken while refactoring for hardhat's compilation pipeline - e.g. even single files aren't excluded with this option anymore.

**Comment by cgewecke** [comment]
This should be fixed with `1.0.3`. There's a usage explanation in the [release notes][1] but the basic idea is:
```js
// Example
excludeContracts: ['folder/', 'file.sol', 'file.sol:contractName'] 
```  

[1]: https://github.com/cgewecke/hardhat-gas-reporter/releases/tag/v1.0.3
